hydrolib/core/utils.py

```diff
@@ -175,3 +176,127 @@ def get_operating_system() -> OperatingSystem:
+
+        Args:
+            file_path (Path): The file path to convert to the OS path style.
+            file_path_style (PathStyle): The file path style of the given file path.
+
+        Returns:
+            str: The converted file path with OS path style.
+
+        Raises:
+            NotImplementedError: When this function is called with a PathStyle other than WINDOWSLIKE or UNIXLIKE.
+        """
+
+        return FilePathStyleConverter._convert(
+            file_path, source_path_style, self._os_path_style
+        )
+
+    def convert_from_os_style(
+        self, file_path: Path, target_path_style: PathStyle
+    ) -> str:
+        """Convert the file path from the path style of the current operating system to the target path style.
+
+        Args:
+            file_path (Path): The file path to convert to the target path style.
+            target_path_style (PathStyle): The target path style to convert the file path to.
+
+        Returns:
+            str: The converted file path with the target path style.
+
+        Raises:
+            NotImplementedError: When this function is called with a PathStyle other than WINDOWSLIKE or UNIXLIKE.
+        """
+
+        return FilePathStyleConverter._convert(
+            file_path, self._os_path_style, target_path_style
+        )
+
+    @classmethod
+    def _convert(
+        cls, file_path: Path, source_path_style: PathStyle, target_path_style: PathStyle
+    ) -> str:
+        if source_path_style == target_path_style:
+            return file_path.as_posix()
+
+        if (
+            source_path_style == PathStyle.UNIXLIKE
+            and target_path_style == PathStyle.WINDOWSLIKE
+        ):
+            return FilePathStyleConverter._from_posix_to_windows_path(file_path)
+        elif (
+            source_path_style == PathStyle.WINDOWSLIKE
+            and target_path_style == PathStyle.UNIXLIKE
+        ):
+            return FilePathStyleConverter._from_windows_to_posix_path(file_path)
+        else:
+            raise NotImplementedError(
+                f"Cannot convert {source_path_style} to {target_path_style}"
+            )
+
+    @classmethod
+    def _from_posix_to_windows_path(cls, posix_path: Path) -> str:
+        is_relative = not posix_path.as_posix().startswith("/")
+
+        if is_relative:
+            return posix_path.as_posix()
+
+        root = posix_path.parts[1]
+        windows_root = root + ":/"
+        parts = posix_path.parts[2:]
+
+        windows_path = windows_root + "/".join(parts)
+
+        return windows_path
+
+    @classmethod
+    def _from_windows_to_posix_path(cls, windows_path: Path) -> str:
+        windows_path_str = str(windows_path).replace("\\", "/")
+        windows_path = Path(windows_path_str)
+
+        root = windows_path.parts[0]
+        is_relative = ":" not in root
+
+        if is_relative:
+            return windows_path_str
+
+        posix_root = "/" + root.split(":")[0] + "/"
```

Comment:
See comment in previous function: better to have user-definable mapping table. (maybe defaulting to your current code when not given by user)